### DISCUSSION TITLE
MaterialInstance to assimp aiMaterial

### DISCUSSION THREAD

**Discussion by Pankaj003** [google/filament#4510] [Q&A]
Hi, 
I have a requirement to save the 3D scene in FBX format and for that I am using assimp library. 
To export the materials I am trying to convert MaterialInstance data structure to assimp aiMaterial .
Have anyone ever tried to do this? Any pointer or tips on how to achieve this will be very helpful and appreciated.

Thanks

**Comment by romainguy** [comment]
Assimp only have limited support for PBR and doesn't support all the features we expose in our materials. That said you can query material parameter values and various other material properties so you should be able to do a "best effort".

**Comment by Pankaj003** [comment]
Hi Romain,
Thanks for your reply. Are the properties mentioned in the link https://google.github.io/filament/Materials.html is full set of material capability?
Where can I find the complete list of supported material params/properties?

Thanks,

**Reply by romainguy** [reply]
The material params are defined by the author of the material (you). Otherwise yes that document describes the full material capabilities.
